python/src/scipp/plot/profiler.py

```diff
@@ -0,0 +1,340 @@
+            self.ax_pick.clear()
+            self.ax_pick.set_ylim([
+                self.params["values"][self.name]["vmin"],
+                self.params["values"][self.name]["vmax"]
+            ])
+        self.profile_viewer = None
+        if self.profile_scatter is not None:
+            self.ax.collections = []
+            self.fig.canvas.draw_idle()
+            del self.profile_scatter
+            self.profile_scatter = None
+
+    def slice_data(self):
+
+        # Run Slicer2d slice_data
+        super().slice_data()
+
+        # Update the position of the slice position indicator
+        for dim, val in self.slider.items():
+            if not val.disabled:
+                if self.slice_pos_rectangle is not None:
+                    new_pos = self.slider_coord[self.name][dim][
+                        dim, val.value].value
+                    self.slice_pos_rectangle.set_x(new_pos)
+                    if self.histograms[self.name][dim][dim]:
+                        self.slice_pos_rectangle.set_width(self.slider_coord[
+                            self.name][dim][dim, val.value + 1].value -
+                                                           new_pos)
+                    else:
+                        new_pos -= 0.5 * self.slice_pos_rectangle.get_width()
+                    self.slice_pos_rectangle.set_x(new_pos)
+
+    def toggle_mask(self, change):
+        super().toggle_mask(change)
+        if self.profile_viewer is not None:
+            self.profile_viewer[self.profile_key].masks[self.name][
+                change["owner"].masks_name].value = change["new"]
+        return
+
+    def compute_profile(self, event):
+        # Find indices of pixel where cursor lies
+        dimx = self.xyrebin["x"].dims[0]
+        dimy = self.xyrebin["y"].dims[0]
+        ix = int((event.xdata - self.current_lims["x"][0]) /
+                 self.image_pixel_size[dimx])
+        iy = int((event.ydata - self.current_lims["y"][0]) /
+                 self.image_pixel_size[dimy])
+        # Resample the 3d cube down to a 1d profile
+        return self.resample_image(self.da_with_edges,
+                                   coord_edges={
+                                       dimy: self.da_with_edges.coords[dimy],
+                                       dimx: self.da_with_edges.coords[dimx]
+                                   },
+                                   rebin_edges={
+                                       dimy: self.xyrebin["y"][dimy,
+                                                               iy:iy + 2],
+                                       dimx: self.xyrebin["x"][dimx, ix:ix + 2]
+                                   })[dimy, 0][dimx, 0]
+
+    def create_profile_viewer(self, prof):
+        # We need to extract the data again and replace with the original
+        # coordinates, because coordinates have been forced to be bin-edges
+        # so that rebin could be used. Also reset original unit.
```

Comment:
Does it actually (have to) force the screen-place coord to bin edges? Rebin is used only for X and Y of the image, which should not be relevant for the profile?

Reply:
Yes, for the profiles we rebin to the size of the currently displayed pixels, with a single bin in both x and y.
Then, if you change which dimension you want as profile by clicking on a button, you will also need the 3rd dim with edges because it has now either become x or y.